ocean_spark/hooks.py

```diff
@@ -177,6 +180,28 @@ def get_app_page_url(self, app_id: str) -> str:
             f"apps/clusters/{self.cluster_id}/apps/{app_id}/overview&accountId={self.account_id}",
         )
 
+    def test_connection(self) -> Tuple[bool, str]:
+        method, path = GET_CLUSTER_ENDPOINT
+        try:
+            response = self._do_api_call(
+                method,
+                path.format(
+                    cluster_id=self.cluster_id,
+                ),
+                {},
+            )
+            if response["response"]["items"][0]["state"] not in [
```

Comment:
should we handle a `Bad Request` response if the cluster is not found, for example?

```json
"response": {
        "status": {
            "code": 400,
            "message": "Bad Request"
        },
        "errors": [
            {
                "code": "RESOURCE_DOES_NOT_EXIST",
                "message": "Cluster not found"
            }
        ]
    }
```

Reply:
or in that case, it returns  `false` in the `except` section?

Reply:
the self._do_api_call already call raise_for_status()
https://3.python-requests.org/user/quickstart/#response-status-codes